### PR TITLE
Add mobile drawer and useBreakpoints hook

### PR DESCRIPTION
**Pull request by abcrane123**
**What changed? Why?**
- add mobile wallet drawer component

https://github.com/user-attachments/assets/66171dd5-e113-4bd8-8eab-79ee435b67bd

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/useBreakpoints.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  xl: '(min-width: 1024px) and (max-width: 1279px)',
+  '2xl': '(min-width: 1280px)',
+};
+
+export function useBreakpoints() {
+  const [currentBreakpoint, setCurrentBreakpoint] = useState<
+    string | undefined
+  >(undefined);
+
+  // handles SSR case where window would be undefined,
```
Comment (Zizzamia) [inline src/useBreakpoints.ts:17]: Loooooove the comments

File: src/useBreakpoints.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+
+// tailwind breakpoints
+const breakpoints = {
```
Comment (Zizzamia) [inline src/useBreakpoints.ts:4]: As this is a constant, is probably better have it as UPPERCASE. But more a nit, honestly.

File: src/wallet/components/WalletBottomSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  useCallback,
+  useMemo,
+} from 'react';
+import { useAccount } from 'wagmi';
+import { Identity } from '../../identity/components/Identity';
+import { background, cn } from '../../styles/theme';
+import type { WalletBottomSheetReact } from '../types';
+import { useWalletContext } from './WalletProvider';
+
+export function WalletBottomSheet({
```
Comment (Zizzamia) [inline src/wallet/components/WalletBottomSheet.tsx:14]: is BottomSheet a common term for those kind of components? just a curiosity.

Comment (abcrane123) [reply src/wallet/components/WalletBottomSheet.tsx:14]: i feel like i've seen `Drawer` or `Tray` used more frequently but `WalletBottomSheet` was in the designs

Comment (Zizzamia) [reply src/wallet/components/WalletBottomSheet.tsx:14]: Mmmm, it's not a public API so it's fine for now, but in general feel free to triple check those names and see what's more common across other Design Systems.

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**